### ISSUE TITLE
alternator test fails [branch-5.2]

### ISSUE THREAD

**Issue by avikivity** [scylladb/scylladb#13239]
https://jenkins.scylladb.com/view/nexts/job/scylla-5.2/job/next/57/testReport/junit/(root)/non-boost%20tests/Unit_Tests___alternator_run_dev_3/

```
=== TEST.PY STARTING TEST alternator.run.3 ===
export UBSAN_OPTIONS='halt_on_error=1:abort_on_error=1:suppressions=/jenkins/workspace/scylla-5.2/next@2/scylla/ubsan-suppressions.supp'
export ASAN_OPTIONS='disable_coredump=0:abort_on_error=1:detect_stack_use_after_return=1'
test/alternator/run --junit-xml=/jenkins/workspace/scylla-5.2/next@2/scylla/testlog/x86_64/dev/xml/alternator.run.3.xunit.xml
=== TEST.PY TEST alternator.run.3 OUTPUT ===
Scylla is: /jenkins/workspace/scylla-5.2/next@2/scylla/build/dev/scylla.
Booting Scylla on 127.1.5.62 in /jenkins/workspace/scylla-5.2/next@2/scylla/testlog/x86_64/dev/scylla-test-1342...
Boot successful (3.4 seconds).
Traceback (most recent call last):
  File "/jenkins/workspace/scylla-5.2/next@2/scylla/test/alternator/run", line 92, in <module>
    cluster.connect().execute("INSERT INTO system_auth.roles (role, salted_hash) VALUES ('alternator', 'secret_pass')")
    ^^^^^^^^^^^^^^^^^
  File "cassandra/cluster.py", line 1733, in cassandra.cluster.Cluster.connect
  File "cassandra/cluster.py", line 1769, in cassandra.cluster.Cluster.connect
  File "cassandra/cluster.py", line 1756, in cassandra.cluster.Cluster.connect
  File "cassandra/cluster.py", line 3572, in cassandra.cluster.ControlConnection.connect
  File "cassandra/cluster.py", line 3617, in cassandra.cluster.ControlConnection._reconnect_internal
cassandra.cluster.NoHostAvailable: ('Unable to connect to any servers', {'127.1.5.62:9042': OperationTimedOut('errors=None, last_host=None')})
```

**Comment by nyh** [comment]
Something odd happened in this test run: We booted Scylla, waited until CQL is available ("available" the Python driver means we can make the CQL connection), and then tried to connect a second time (to set up Alternator's authentication, using CQL unfortunately) - and this timed out. I don't know how to explain it.

The particular connect() call uses the driver's default timeout , which is just 2 or 5 seconds (there are two different timeouts involved), which isn't much (during the test itself we increased it to 60 seconds, following #11289) - but still 2 seconds during a dev build (it's not even a debug build) is quite a lot, and this is the first time I ever saw it not being enough.

If this is just a case of "2 seconds not being enough" I can easily increase the timeout in that code, but I wonder if it can be something more serious... (if it is, it has nothing to do with the Alternator API, because what timeout out was a CQL connection attempt, not Alternator).
